com.unity.render-pipelines.universal/Shaders/LitDisplacement.hlsl

```diff
@@ -0,0 +1,44 @@
+#ifndef UNIVERSAL_LIT_DISPLACEMENT_INCLUDED
+#define UNIVERSAL_LIT_DISPLACEMENT_INCLUDED
+
+// Return view direction in tangent space, make sure tangentWS.w is already multiplied by GetOddNegativeScale()
+half3 GetViewDirectionTangentSpace(half4 tangentWS, half3 normalWS, half3 viewDirWS)
```

Comment:
Why not use TransformWorldToTangent? https://github.com/Unity-Technologies/Graphics/blob/master/com.unity.render-pipelines.core/ShaderLibrary/SpaceTransforms.hlsl#L180

Reply:
The main reason is that TransformWorldToTangent is more expensive, because they do matrix inversion to find from TangentToWorld matrix. Which is needed for correct non-uniform scale.
However builtin uses simple transposed TangentToWorld matrix, so I decided to go this path to save performance (Also we use it in shadergraph apperently com.unity.shadergraph\Editor\Generation\Templates\BuildSurfaceDescriptionInputs.template.hlsl).
It depends if we want to sacrife performance here.

Also other strange thing I noticed that we multiply GetOddNegativeScale() with tangent sign in vertex shader, where in shadergraph we do this only when needed. This makes most of the SpaceTranform.hlsl functions not usable, as they do GetOddNegativeScale() multiply again (making it two times).

LitForwardPass.hlsl:
    real sign = input.tangentOS.w * GetOddNegativeScale();
    output.tangentWS = half4(normalInput.tangentWS.xyz, sign);

com.unity.render-pipelines.universal\Editor\ShaderGraph\Includes\Varyings.hlsl
    float4 tangentWS = float4(TransformObjectToWorldDir(input.tangentOS.xyz), input.tangentOS.w);

Reply:
Abou the duplicated OddNegativeScale, if it's applying twice then it's a bug we should solve. Maybe you can report it so we keep track to fix later.